### DISCUSSION TITLE
toggle detail information of a mapped list item individually

### DISCUSSION THREAD

**Discussion by esrefbozkurt** [neuefische/questions#403] [Web] [answered]
Hey there,

I tried to toggle the description component inside my Exercise component. I'm using a show useState and a toggleId useState to be able to toggle every single Exercise Card. It works but just on the second Click. Pls help 🙏🏽

here is my [PR ](https://github.com/esrefbozkurt/capstone-project/pull/11)

<img width="665" alt="Screenshot 2022-12-15 at 13 11 07" src="https://user-images.githubusercontent.com/113592929/207856122-44c7b27c-243a-4cf3-82e6-3466c5656fca.png">

**Comment by actyralf** [comment]
`show` is a state variable and state variables don't update immediately after calling their setter. They will only update on next render. So if you evaluate `show` (line 14: `if(show)`), it will still have the value it had before your call to `setShow` in line 13. So a quick fix would be to check for the opposite (`if !show`) - this would work but would make your code hard to read and understand. I am not sure what you want to do with this variable `show` at all as you don't use it in your JSX. Are you sure you need this variable at all?

**Comment by julialoeschel** [comment, marked as answer]
We discusses your code, as Ralf sais: you dont need the show state. you need to check for `!toggleIds.includes(id)` befor setting yout toggleId State. 

The rest should work like that.
🐜
